src/image/p5.Image.js

```diff
@@ -825,7 +886,261 @@ p5.Image.prototype.isModified = function() {
  *
  */
 p5.Image.prototype.save = function(filename, extension) {
-  p5.prototype.saveCanvas(this.canvas, filename, extension);
+  if (this.gifProperties) {
+    p5.prototype.saveGif(this, filename);
+  } else {
+    p5.prototype.saveCanvas(this.canvas, filename, extension);
+  }
+};
+
+// GIF Section
+/**
+ * Starts an animated GIF over at the beginning state.
+ *
+ * @method reset
+ * @example
+ * <div><code>
+ * let gif;
+ *
+ * function preload() {
+ *   gif = loadImage('assets/animated-counting.gif');
+ * }
+ *
+ * function draw() {
+ *   background(255);
+ *   image(gif, 0, 0);
+ * }
+ *
+ * function mousePressed() {
+ *   // Click to reset the GIF
+ *   gif.reset();
+ * }
+ * </code></div>
+ * @alt
+ * image of a GIF that counts from 0 - 9 repeatedly, changing
+ * every second. When you click it starts back over at 0.
+ *
+ */
+p5.Image.prototype.reset = function() {
+  if (this.gifProperties) {
+    var props = this.gifProperties;
+    props.playing = true;
+    props.timeSinceStart = 0;
+    props.timeDisplayed = 0;
+    props.loopCount = 0;
+    props.displayIndex = 0;
+    this.drawingContext.putImageData(props.frames[0], 0, 0);
+  }
+};
+
+/**
+ * Gets the index for the frame that is currently visible in an animated GIF.
+ *
+ * @method getCurrentFrame
+ * @return {Number}       The index for the currently displaying frame in animated GIF
+ * @example
+ * <div><code>
+ * let gif;
+ *
+ * function preload() {
+ *   gif = loadImage('assets/animated-counting.gif');
+ * }
+ *
+ * function setup() {
+ *   noStroke();
+ * }
+ *
+ * function draw() {
+ *   background(255);
+ *   let frame = gif.getCurrentFrame();
+ *   if (frame % 2 === 0) {
+ *     fill(149, 66, 244);
+ *   } else {
+ *     fill(106, 225, 140);
+ *   }
+ *   image(gif, 0, 0);
+ *   ellipse(frame * 10, height / 2, 40);
+ * }
+ * </code></div>
```

Comment:
I like this, but this example is a little complex to follow. Could you add one more that just displays the current frame number using the text() function overtop of the drawn gif?